### PR TITLE
#5077 feat: Refactor Api class to accept configProvider

### LINKED ISSUE #5077: Refactor Api class to accept `configProvider`
Well, these `REQ` and `RES` keep popping up because this `Api` class was meant to be generic, but now this generic class is supposed to know about certain "attester"s. That's too specific and contradicts the initial design of this class.
So it'd be better for this class to query some other injected function or interface for additional handlers. This function or object can be injected in the constructor.

_Originally posted by @rrrooommmaaa in https://github.com/FlowCrypt/flowcrypt-browser/pull/5067#discussion_r1162907991_

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR refactored Api class to accept configProvider

close #5077 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by ioanmo226** [comment]
This PR also leet KeysOpenPgpOrg to use KeysOpenPgpOrgConfig

File: test/source/mock/keys-openpgp-org/keys-openpgp-org-endpoints.ts [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
-    if (email === 'test.only.pubkey.keys.openpgp.org@allowed-domain.test' || email === 'test.only.pubkey.keys.openpgp.org@other.com') {
-      return somePubkey;
-    }
-    throw new HttpClientErr('Pubkey not found', 404);
-  },
+export type KeysOpenPGPOrgConfig = Record<string, string>;
+
+export const getMockKeysOpenPGPOrgEndpoints = (keysOpenPGPOrgConfig: KeysOpenPGPOrgConfig | undefined): HandlersDefinition => {
+  return {
+    '/keys-openpgp-org/vks/v1/by-email/?': async ({}, req: IncomingMessage) => {
```
Comment (sosnovsky) [inline test/source/mock/keys-openpgp-org/keys-openpgp-org-endpoints.ts:12]: probably `: IncomingMessage` can be removed here

**Review by sosnovsky — COMMENTED** [review]
Looks great 👍

File: test/source/mock/attester/attester-endpoints.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { expect } from 'chai';
 import { Dict } from '../../core/common';
 import { Util } from '../../util';
 
-interface PubKeyLookUpResult {
+export interface PubKeyLookUpResult {
```
[the remark this answers is not in the document]
Comment (ioanmo226) [reply test/source/mock/attester/attester-endpoints.ts:11]: In fact, tried to use this interface in `KeysOpenPgpOrgConfig` but realized we don't need `returnError`, `delayInSeconds` for this.

**Review by sosnovsky — APPROVED** [review]

**Pull request merged**